examples/pycaffe/layers/pascal_multilabel_datalayers.py

```diff
@@ -0,0 +1,262 @@
+# imports
+import json, time, pickle, scipy.misc, skimage.io, caffe
+
+import numpy as np
+import os.path as osp
+
+from xml.dom import minidom
+from random import shuffle
+from threading import Thread
+from PIL import Image
+
+from pascal_multilabel_with_datalayer_tools import SimpleTransformer
```

Comment:
Now that you mention it, I might have gotten a bit carried away.

I was thinking about it this morning again, and I don't think a base class is necessarily required, but I would still think that using the `BatchLoader` in both the sync and async cases is a good idea as it clarifies what the steps in the process are, and users can "drill down" further if they need more info as to what actually happens when loading an image.